hw_diag/utilities/security.py

```diff
@@ -0,0 +1,17 @@
+import gnupg
+
+gpg = gnupg.GPG()
+
+
+def get_verified_data(data):
+    """ Verify the signature on the contents of the string 'data'
+        Return values:
+            Verified data - if succeed
+            None - if verification failed
+    """
+    verified = gpg.verify(data)
+    if verified.valid:
```

Comment:
I was wrong about this. `trust_text` is for something else. The old way you had it was fine. We do have to use `decrypt` in either case. That is the part I think is most confusing.

Could you add a comment to explain that this decryption is just separating the payload from the rest of the signature and not doing any real decryption because the message is sent in cleartext?

Reply:
Sure